code/GlobalNavSiteTreeExtension.php

```diff
@@ -8,8 +8,26 @@ class GlobalNavSiteTreeExtension extends DataExtension {
 	);
 
 
+	/**
+	 * Gets the hostname
+	 *
+	 * @return string Protocol and hostname only
+	 */
 	public static function get_toolbar_hostname() {
-		return Config::inst()->get('GlobalNav','use_localhost') ? Director::absoluteBaseURL() : Config::inst()->get('GlobalNav','hostname');
+		return Config::inst()->get('GlobalNav','use_localhost')
+			? Director::protocolAndHost()
```

Comment:
What happens if the site url is http://localhost/mysite? That would break https://github.com/silverstripe/silverstripe-globaltoolbar/pull/25/files#diff-995d606ee3473545de2ad7aa9bf808ccR51 right?